### PR TITLE
Changes amp-story-bookend config to use camelcase

### LINKED ISSUE #15893: Consider switching bookend 1.0 config to camel case instead of dashes.
We should consider switching the bookend 1.0 config to using camel case instead of dashes.

Most AMP components use camel case for their JSON config:

``amp-consent``

````
{
  "checkConsentHref": "https://example.com/api/show-consent",
  "promptUI": "consent-ui"
}
````

``amp-analytics``

````
{
  "requests": {
    "pageview": "https://foo.com/pixel?RANDOM"
  },
  "triggers": {
    "trackPageview": {
      "on": "visible",
      "request": "pageview"
    }
  }
}
````

etc...

### PR DESCRIPTION
**Pull request by Enriqe**
Fixes #15893

### CODE REVIEW

**Comment by Enriqe** [comment]
We should also set a reminder to update the docs when this goes live.

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 1]
```diff
@@ -319,6 +319,24 @@ export class AmpStoryBookend extends AMP.BaseElement {
     this.toggleDesktopAttribute_(isDesktop);
   }
 
+  /**
+   * Reads the bookend version from the bookend JSON config.
+   * @param {!JsonObject} config
+   * @return {?Array<string|!JsonObject>} Providers config.
+   */
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Please use ``@private``

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 2]
```diff
@@ -319,6 +319,24 @@ export class AmpStoryBookend extends AMP.BaseElement {
...
+    const bookendVersion = config[BOOKEND_VERSION_KEY];
+    if (bookendVersionDeprecated) {
+      user().warn('amp-story-bookend config', '`bookend-version` is ' +
+      'deprecated, please use `bookendVersion`');
+      return bookendVersionDeprecated;
+    } else if (bookendVersion) {
+      return bookendVersion;
+    }
+    return null;
+  }
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Readability nit:
````
if (config['bookend-version']) {
  user().warn(....);
}

return config[BOOKEND_VERSION_KEY] || config['bookend-version'] || null;
````

File: extensions/amp-story/1.0/amp-story-share.js [hunk 3]
```diff
@@ -306,6 +300,24 @@ export class ShareWidget {
...
+    const shareProviders = config['shareProviders'];
+    if (shareProvidersDeprecated) {
+      user().warn('amp-story-bookend config', '`share-providers` is ' +
+        'deprecated, please use `shareProviders`.');
+      return shareProvidersDeprecated;
+    } else if (shareProviders) {
+      return shareProviders;
+    }
+    return null;
+  }
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-share.js]: Readability nit (see below)

File: extensions/amp-story/1.0/amp-story-share.js [hunk 4]
```diff
@@ -314,7 +326,7 @@ export class ShareWidget {
     this.loadRequiredExtensions();
 
     this.requestService_.loadBookendConfig().then(config => {
-      const providers = config && config[SHARE_PROVIDERS];
+      const providers = config && this.readProviders(config);
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-share.js]: What do you think of simply doing 
````
const providers = config && config[SHARE_PROVIDERS] || config['share-providers'];
````
here, so you don't have to write a bookend error message in the share widget file?
The bookend error message should be moved to the other class, with the bookend version warning.

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js (hunk 2 above)
Comment (newmuis) [reply extensions/amp-story/1.0/bookend/amp-story-bookend.js]: +1, also maybe with:

```javascript
const DEPRECATED_BOOKEND_VERSION_KEY = 'bookend-version';```

File: extensions/amp-story/1.0/amp-story-share.js (hunk 4 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/amp-story-share.js]: ok, only thing is that this way we won't show the warning when they use the 'share-providers' key. Just making sure we are ok with that

**Comment by Enriqe** [comment]
@newmuis ready to merge

**Pull request merged**